core/rawdb/freezer_test.go

```diff
@@ -0,0 +1,85 @@
+package rawdb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// Quorum
+
+func Test_freezer_shouldClose_whenHaveSubscribers(t *testing.T) {
+	// given
+	mockFreezer := newFreezerMock(t)
+
+	// when
+	err := mockFreezer.Close()
+
+	// then
+	assert.Nil(t, err)
+}
+
+func Test_shouldCloseGracefully_whenNoSubscribers(t *testing.T) {
```

Comment:
MINOR: I don't understand difference between `whenNoSubscribers` and `whenHaveSubscribers`. 

Honestly, I think you can gather everything under one test `TestClose`

Reply:
if the channel is closed, a panic will be raised.

In this case, is not an issue of closed it seems. is that there is no one receiving in the other side of the channel (subscribers).

so I'm testing both scenarios: the happy path and the path where closed is called twice.

However, it can be merged into one but then is not specific for each case (as the given is different). I can merge it together and add comments to explicitly explain the steps

Reply:
changed to 1 test